core/kernel/persistence/newsql/NewSqlRdf.php

```diff
@@ -65,14 +70,39 @@ public function add(core_kernel_classes_Triple $triple)
     }
 
     /**
+     * Add id to set of triple values. Put id in first position to match parameter types
+     *
      * @param core_kernel_classes_Triple $triple
      * @return array
      * @throws Exception
      */
     protected function tripleToValue(core_kernel_classes_Triple $triple)
     {
-        $values = parent::tripleToValue($triple);
-        $values['id'] = $this->getUniquePrimaryKey();
-        return $values;
+        return ['id' => $this->getUniquePrimaryKey()] + parent::tripleToValue($triple);
+    }
+
+    /**
+     * @return array
+     */
+    protected function getTripleParameterTypes()
+    {
+        return [
+            // id
+            ParameterType::STRING,
+            // modelid
+            ParameterType::INTEGER,
+            // subject
+            ParameterType::STRING,
+            // predicate
+            ParameterType::STRING,
+            // object
+            ParameterType::STRING,
+            // l_language
+            ParameterType::STRING,
+            // author
+            ParameterType::STRING,
+            // epoch
+            ParameterType::STRING,
+        ];
```

Comment:
We can avoid maintaining the column schema on both places for other columns:

```suggestion
        return array_merge(
            [
                // id
                ParameterType::STRING
            ], 
            parent::getTripleParameterTypes()
        );
```

Reply:
good catch